### PR TITLE
Non-Scalar default dangerous change: Fix for #1566

### PR DESCRIPTION
**Pull request by mjmahone**
@helloqiu identified in #1566 that our dangerous changes check was incorrectly saying any non-scalar default value was a dangerous change, even if the default value hadn't actually changed. This is because we need to check the default value change for deep equality, rather than just whether the objects are identical.

### CODE REVIEW

File: src/utilities/findBreakingChanges.js [hunk 1]
```diff
@@ -496,6 +498,41 @@ function isChangeSafeForInputObjectFieldOrFieldArg(
...
+    if (Array.isArray(oldDefaultValue) !== Array.isArray(newDefaultValue)) {
+      return false;
+    }
+
+    for (const key in oldDefaultValue) {
+      if (
+        !isChangeSafeForDefaultValue(oldDefaultValue[key], newDefaultValue[key])
+      ) {
+        return false;
+      }
```
Comment (IvanGoncharov) [inline src/utilities/findBreakingChanges.js]: @mjmahone According to current implementation adding properties to object is safe.
For example `isChangeSafeForDefaultValue({}, {a: 1}) => true` or `isChangeSafeForDefaultValue({a: undefined}, {a: 1}) => true`.

**Comment by IvanGoncharov** [comment]
@mjmahone I think it's better to add `isDeepEqual` to `jsutils` folder.
We can use https://github.com/substack/node-deep-equal as basis and remove support for all weird types (e.g. `Buffer`, `arguments`, etc.)

File: src/utilities/findBreakingChanges.js (hunk 1 above)
Comment (Neitsch) [reply src/utilities/findBreakingChanges.js]: I second this. Can we add a couple more testcases to make sure that check works correctly?

Comment (mjmahone) [reply src/utilities/findBreakingChanges.js]: That should be correct: adding a previously undefined value is OK, but removing a previously undefined value is not (recursively).

**Comment by mjmahone** [comment]
This will not make it into 14.1 milestone: there's a little bit too much clean-up work required here. I'll try to fast-follow with a patch to get this included, but I don't want to hold up the 14.1 release on a bugfix.

**Pull request closed without merging**